pageserver/src/walredo.rs

```diff
@@ -1056,11 +1060,16 @@ impl Drop for NoLeakChild {
             Some(child) => child,
             None => return,
         };
+        let tenant_id = self.tenant_id;
         // Offload the kill+wait of the child process into the background.
         // If someone stops the runtime, we'll leak the child process.
         // We can ignore that case because we only stop the runtime on pageserver exit.
         BACKGROUND_RUNTIME.spawn(async move {
             tokio::task::spawn_blocking(move || {
+                // Intentionally don't inherit the tracing context from whoever is dropping us.
+                // This thread here is going to outlive of our dropper.
+                let span = tracing::info_span!("NoLeakChild::drop", %tenant_id);
```

Comment:
I think it would be natural for the span to be caught from the place of creating a drop guard, but in our case, it would probably the tenant loading or creation or attach, which we don't really need either.

I don't enjoy the name, but I guess it's fine. Perhaps `walredo{tenant_id}` would work?

Reply:
> I think it would be natural for the span to be caught from the place of creating a drop guard, but in our case, it would probably the tenant loading or creation or attach, which we don't really need any of those.

Yeah, that's why I decided to start a new one here.

> I don't enjoy the name, but I guess it's fine. Perhaps walredo{tenant_id} would work?

Hm, just noticed that `:` in a span name is probably a bad idea given how our logging formats the spans.

Yeah, `walredo` is fine.